src/pytest_servers/gcs.py

```diff
@@ -0,0 +1,54 @@
+import logging
+import time
+
+import pytest
+import requests
+
+from .utils import get_free_port
+
+logger = logging.getLogger(__name__)
+
+GCS_DEFAULT_PORT = 4443
+
+
+@pytest.fixture(scope="session")
+def fake_gcs_server(docker_client):
+    if docker_client is None:
+        logger.warning(
+            "fake-gcs-server cannot run because docker is not available."
+        )
+        yield None
+        return
+
+    # Some features, such as signed URLs and resumable uploads, require
```

Comment:
When initiating a resumable upload, initial call that initiates the upload returns the URL that needs to be POSTed to upload each chunk. The resumable uploads endpoint in fake-gcs-server returns as URL whatever is provided using `-external-url`, falling back to the default value of localhost:4443 if that's not defined, breaking uploads if running on a custom port.

Seee https://github.com/fsouza/fake-gcs-server/blob/f36425ff00b9bdd12b65aebe40b2ebaf2f2dd3f3/fakestorage/server.go#L85